src/server/lock_manager/deadlock.rs

```diff
@@ -224,7 +224,8 @@ const LEADER_KEY: &[u8] = b"";
 
 /// Returns true if the region containing the LEADER_KEY.
 fn is_leader_region(region: &'_ Region) -> bool {
-    region.get_start_key() <= LEADER_KEY
+    !region.get_peers().is_empty()
+        && region.get_start_key() <= LEADER_KEY
         && (region.get_end_key().is_empty() || LEADER_KEY < region.get_end_key())
```

Comment:
Could we remove L228?

Reply:
It's not necessary, but it's an idiom to check key in range.